erts/emulator/beam/erl_process.c

```diff
@@ -9006,11 +9067,11 @@ erts_internal_suspend_process_2(BIF_ALIST_2)
         if (rp == ERTS_PROC_LOCK_BUSY)
             send_sig = !0;
         else {
-            send_sig = !suspend_process(BIF_P, rp);
+            send_sig = !suspend_process(BIF_P, rp, pause_proc_timer);
```

Comment:
Pause the timer before or after calling `suspend_process()`. Here you are in the context where you can do it right away.